packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -510,10 +510,13 @@ export class BaseResolversVisitor<
     const nestedMapping: { [typeName: string]: boolean } = {};
     const typeNames = this._federation.filterTypeNames(Object.keys(allSchemaTypes));
 
-    typeNames.forEach(typeName => {
-      const schemaType = allSchemaTypes[typeName];
-      nestedMapping[typeName] = this.shouldMapType(schemaType, nestedMapping);
-    });
+    // avoid checking all types recursively if we have no `mappers` defined
```

Comment:
nice! :) 
in cases where `mappers` is defined, do you think we can optimize something within `shouldMapType`? 
I guess the result of `shouldMapType` can be cached for a given type? so we can only check it once? 🤔

Reply:
`shouldMapType()` has already a cache system that relies on `checkedBefore` parameter
However, we could try to have `nestedMapping` (in `createResolversFields()`) moved as an instance variable, however, I'm not sure of the side effect...

I'll make the changes and test!

Reply:
@dotansimha tested on a real project, no regression, and slightly better perf when `mappers` is defined.